### PR TITLE
Features/react flow

### PR DESCRIPTION
**Pull request by akinolu52**
New Implementation for the graph panel

### CODE REVIEW

File: re_data_ui/src/components/FlowGraph.tsx [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+  );
+
+  const getAllOutgoer = (node: Node, el: Elements): Node[] => getOutgoers(node, el).reduce(
+    (memo: any, outgoer) => [
+      ...memo,
+      outgoer,
+      ...getAllOutgoer(outgoer, el),
+    ],
+    [],
+  );
```
Comment (dejii) [inline re_data_ui/src/components/FlowGraph.tsx:57]: I see this does a reduce operation in a recursive manner, we should discuss this in full detail.

File: re_data_ui/src/components/FlowGraph.tsx [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+
+      setElements((prevElements) => prevElements?.map((elem) => {
+        const element = elem;
+        const incomerIds = allIncomer.map((i) => i.id);
+        const outgoerIds = allOutgoer.map((o) => o.id);
+
+        if (isNode(element)) {
+          const highlight = element.id === node.id
+              || incomerIds.includes(element.id)
+            || outgoerIds.includes(element.id);
```
Comment (dejii) [inline re_data_ui/src/components/FlowGraph.tsx:90]: A set would be more efficient here when there are a large amount of nodes.

File: re_data_ui/src/components/FlowGraph.tsx [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+} from 'react-flow-renderer';
+import { useSearchParams } from 'react-router-dom';
+import '../graph.css';
+import { getLayoutElements } from '../utils';
+import CustomNode from './CustomNode';
+
+export interface Props {
+  data: Elements;
+  disableClick?: boolean;
+}
```
[the remark this answers is not in the document]
Comment (akinolu52) [reply re_data_ui/src/components/FlowGraph.tsx:19]: okay, fixed

File: re_data_ui/src/components/FlowGraph.tsx (hunk 1 above)
Comment (akinolu52) [reply re_data_ui/src/components/FlowGraph.tsx:57]: funny enough @mateuszklimek  said it's not needed,

the idea is to get all the edges linked to it (its direct/indirect parents and children)

but i'll refactor it to get only its direct children.

File: re_data_ui/src/components/FlowGraph.tsx (hunk 2 above)
Comment (akinolu52) [reply re_data_ui/src/components/FlowGraph.tsx:90]: nice

**Review by dejii — APPROVED** [review]

**Pull request merged**